File: R/KNN.R
library(caret)

# Load 'Iris' dataset
data(iris)

# Split dataset into training and test sets
set.seed(42)
trainIndex <- createDataPartition(iris$Species, p = 0.7, list = FALSE)
trainData <- iris[trainIndex, ]
testData <- iris[-trainIndex, ]

# Create and train a KNN classifier
model <- train(
  Species ~ ., 
  data = trainData, 
  method = "knn",
  trControl = trainControl(method = "none"),
  tuneGrid = data.frame(k = 5),
  metric = "Accuracy"
)

File: Python/KNN.py
from sklearn import datasets
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier

# Load 'Iris' dataset
iris = datasets.load_iris()
X = iris.data
y = iris.target

# Split dataset into training and test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# Create and train a KNN classifier
clf = KNeighborsClassifier(
    n_neighbors=5,
    weights='uniform',
    algorithm='auto',
    leaf_size=30,
    p=2,
    metric='minkowski',
    n_jobs=None  # Use default value for comparability
)
clf.fit(X_train, y_train)